### PR TITLE
chore: use `toUpperCase` in `detectOS`

### PR DESCRIPTION
**Pull request by CodingRule**
Instead of using the switch statement, we use a faster way with an object. Hope it's not useless tho.

### CODE REVIEW

**Comment by CodingRule** [comment]
Also I can change it at utils/getNodeDownloadUrl.ts too

File: apps/site/util/detectOS.ts [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
-      return 'OTHER';
-  }
+  const osMatch = userAgent?.match(/(Win|Mac|Linux|AIX)/);
+  const osMap: Record<string, UserOS> = {
+    Win: 'WIN',
+    Mac: 'MAC',
+    Linux: 'LINUX',
+    AIX: 'AIX',
+  };
+  return osMap[osMatch?.[1] ?? ''] || 'OTHER';
```
Comment (avivkeller) [inline apps/site/util/detectOS.ts]: if we are already matching, why not just use toUpperCase if we find a match, instead of a map?

File: apps/site/util/detectOS.ts [hunk 2]
```diff
@@ -1,21 +1,16 @@
...
 };
 
-// Since `navigator.appVersion` is deprecated, we use the `userAgent``
+// Since navigator.appVersion is deprecated, we use the userAgent
```
Comment (avivkeller) [inline apps/site/util/detectOS.ts]: Please revert this change

Comment (CodingRule) [reply apps/site/util/detectOS.ts]: ok!

Comment (CodingRule) [reply apps/site/util/detectOS.ts]: I did revert it I hope It updated.

File: apps/site/util/detectOS.ts (hunk 1 above)
Comment (CodingRule) [reply apps/site/util/detectOS.ts]: 1 sec Ill be doing that rn

**Comment by CodingRule** [comment]
@RedYetiDev Done Thanks for pointing out to me that I can directly do it with uppercase and I have updated it in your sayings. Thanks! Hope this contribution will help.

File: apps/site/util/detectOS.ts [hunk 3]
```diff
@@ -1,21 +1,12 @@
...
-      return 'AIX';
-    default:
-      return 'OTHER';
-  }
+  // Match OS names and convert to uppercase directly if there's a match
+  const osMatch = userAgent?.match(/(Win|Mac|Linux|AIX)/);
+  return osMatch ? (osMatch[1].toUpperCase() as UserOS) : 'OTHER';
 };
 
+
```
Comment (avivkeller) [inline apps/site/util/detectOS.ts]:
```suggestion
```

**Review by avivkeller — APPROVED** [review]
See my one comment, then LGTM

**Comment by CodingRule** [comment]
Haha thanks! I saw that one coming tbh (=

**Comment by CodingRule** [comment]
> See my one comment, then LGTM

thanks good sir!

**Comment by CodingRule** [comment]
also quick question @RedYetiDev would I be able to apply the same logic here too?

utils/getNodeDownloadUrl.ts too

**Comment by avivkeller** [comment]
In a seperate PR, preferably

**Comment by CodingRule** [comment]
yes ofc

**Review by aymen94 — APPROVED** [review]

**Comment by avivkeller** [comment]
Thanks for your contribution, @CodingRule! It's be merged shortly.

**Comment by CodingRule** [comment]
Ty!
